homeassistant/components/jellyfin/media_player.py

```diff
@@ -1,363 +1,280 @@
+                via_device=(DOMAIN, coordinator.server_id),
+            )
+        else:
+            self._attr_device_info = None
+            self._attr_has_entity_name = False
+            self._attr_name = self.device_name
 
-        for session in self.sessions:
-            if session["Id"] not in self._known:
-                self._known.add(session["Id"])
-                entity = JellyfinMediaPlayer(self, self._hass, self._client, session)
-                self._async_add_entities([entity], False)
+        self._update_from_session_data()
 
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        self.session_data = (
+            self.coordinator.data.get(self.session_id)
+            if self.coordinator.data is not None
+            else None
+        )
+        self.now_playing = (
+            self.session_data.get("NowPlayingItem")
+            if self.session_data is not None
+            else None
+        )
+        self.play_state = (
+            self.session_data.get("PlayState")
+            if self.session_data is not None
+            else None
+        )
+        self._update_from_session_data()
+        super()._handle_coordinator_update()
 
-class JellyfinMediaPlayer(CoordinatorEntity, MediaPlayerEntity):
-    """Represents a Jellyfin Player device."""
+    @callback
+    def _update_from_session_data(self) -> None:
+        """Process session data to update entity properties."""
+        state = None
+        media_content_type = None
+        media_content_id = None
+        media_title = None
+        media_series_title = None
+        media_season = None
+        media_episode = None
+        media_album_name = None
+        media_album_artist = None
+        media_artist = None
+        media_track = None
+        media_duration = None
+        media_position = None
+        media_position_updated = None
+        volume_muted = False
+        volume_level = None
+
+        if self.session_data is not None:
+            state = MediaPlayerState.IDLE
+            media_position_updated = (
+                parse_datetime(self.session_data["LastPlaybackCheckIn"])
+                if self.now_playing
+                else None
+            )
 
-    def __init__(
-        self,
-        coordinator: JellyfinMediaPlayerCoordinator,
-        hass: HomeAssistant,
-        client: JellyfinClient,
-        session: Any,
-    ) -> None:
-        """Initialize the MediaPlayer Entity.
+        if self.now_playing is not None:
+            state = MediaPlayerState.PLAYING
+            media_content_type = CONTENT_TYPE_MAP.get(self.now_playing["Type"], None)
+            media_content_id = self.now_playing["Id"]
+            media_title = self.now_playing["Name"]
+            media_duration = int(self.now_playing["RunTimeTicks"] / 10000000)
+
+        if self.now_playing is not None and media_content_type == MediaType.EPISODE:
+            # HA frontend does not display all info for episode content type
```

Comment:
Comment feels ambiguous. Do we not set everything, because the Frontend won't display it, or is it a note that the frontend won't show everything set here?

Reply:
so we map JF Episode to MediaType.EPISODE but frontend only displays extended series name etc in if type is MediaType.TVSHOW. I'll revise that was more a mental note.